go/tasks/hive/execution_state.go

```diff
@@ -0,0 +1,306 @@
+package hive
+
+import (
+	"context"
+	"fmt"
+	idlCore "github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/plugins"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/utils"
+	"strconv"
+
+	"github.com/lyft/flyteplugins/go/tasks/errors"
+	"github.com/lyft/flyteplugins/go/tasks/hive/client"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core"
+	"github.com/lyft/flytestdlib/logger"
+	utils2 "github.com/lyft/flytestdlib/utils"
+	"time"
+)
+
+type ExecutionPhase int
+
+const (
+	PhaseNotStarted ExecutionPhase = iota
+	PhaseQueued      // resource manager token gotten
+	PhaseSubmitted   // Sent off to Qubole
+
+	PhaseQuerySucceeded
+	PhaseQueryFailed
+)
+
+func (p ExecutionPhase) String() string {
+	switch p {
+	case PhaseNotStarted:
+		return "PhaseNotStarted"
+	case PhaseQueued:
+		return "PhaseQueued"
+	case PhaseSubmitted:
+		return "PhaseSubmitted"
+	case PhaseQuerySucceeded:
+		return "PhaseQuerySucceeded"
+	case PhaseQueryFailed:
+		return "PhaseQueryFailed"
+	}
+	return "Bad Qubole execution phase"
+}
+
+type ExecutionState struct {
+	Phase ExecutionPhase
+
+	// This ID is the cache key and so will need to be unique across all objects in the cache (it will probably be
+	// unique across all of Flyte) and needs to be deterministic.
+	// This will also be used as the allocation token for now.
+	Id string `json:"id"`
+
+	// This will store the command ID from Qubole
+	CommandId string `json:"command_id,omitempty"`
+
+	// This number keeps track of the number of failures within the sync function. Without this, what happens in
+	// the sync function is entirely opaque. Note that this field is completely orthogonal to Flyte system/node/task
+	// level retries, just errors from hitting the Qubole API, inside the sync loop
+	SyncQuboleApiFailures int `json:"sync_qubole_api_failures,omitempty"`
```

Comment:
if we are using json, I prefer to have small names please. `sync-failures` `create-failures`?
Also omitempty works with pointers, you do not need it here as 0 is valid